### PR TITLE
Add a Diagnostics container checking the existence of containers

### PR DESCRIPTION
**Pull request by davisein**
The diagnostics are accessible via /diagnostics/ on the Nginx container.

It simply checks that the pod has internet connection (to download the charts), and that the replication controllers are in place.

The tests can be run via `py.test tests/diagnostics`

### CODE REVIEW

**Comment by davisein** [comment]
@efrenrey

File: src/diagnostics/diagnostics.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+@coroutine
+def _check_kubernetes_status(settings):
+    try:
+        data = yield _get_json(settings, '')
+    except (IOError, tornado.httpclient.HTTPError) as e:
+        raise Return(status_error(unicode(e)))
+
+    if 'paths' not in data or '/api/v1' not in data['paths']:
+        raise Return(status_error)
```
Comment (EfrenRey) [inline src/diagnostics/diagnostics.py]: ```status_error``` is a function. Please also pass a ```msg``` parameter with error description.

Comment (davisein) [inline src/diagnostics/diagnostics.py]: Oops, good catch, added tests to check for it too.

File: src/diagnostics/diagnostics.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        response = yield client.fetch(settings['check_connectivity_url'])
+    except (IOError, tornado.httpclient.HTTPError) as ex:
+        raise Return(status_error(
+            'Requesting "{}" failed: "{}"'.format(settings['check_connectivity_url'], unicode(ex))))
+    except Exception as ex:
+        logger.exception('Exception detected, %s', type(ex))
+        raise Return(status_error(unicode(ex)))
+
+    if response.code != 200:
+        raise Return(status_error('"{}" responded error status code'))
```
Comment (EfrenRey) [inline src/diagnostics/diagnostics.py]: It seems that ```.format()``` is missing

File: src/diagnostics/diagnostics.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+    start_background_checks(settings, system_status, rc_names)
+
+    statics_path = os.path.join(os.path.dirname(__file__), 'assets')
+    application = create_application(system_status, statics_path, bool(os.getenv('DEBUG')))
+    server = tornado.httpserver.HTTPServer(application)
+
+    socket = tornado.netutil.bind_unix_socket("/var/run/elastickube-diagnostics.sock", mode=0777)
+    server.add_socket(socket)
+    if os.getenv('DEBUG'):
```
Comment (albertoarias) [inline src/diagnostics/diagnostics.py]: Please remove this. Use the socks file.

File: src/diagnostics/diagnostics.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+    level = 'WARNING' if not os.getenv('DEBUG') else 'DEBUG'
+    logging.basicConfig(level=level)
+
+    logger.info('Starting server')
+    tornado.netutil.Resolver.configure('tornado.netutil.ThreadedResolver', num_threads=10)
+
+    settings = {}
+    settings_from_env(settings, os.environ)
+
+    rc_names = (
```
Comment (albertoarias) [inline src/diagnostics/diagnostics.py]: Use replication_controllers instead

**Comment by albertoarias** [comment]
@davisein Missing the favicon.

Also we should redirect to this page on 500

**Comment by paulino-corujo** [comment]
ACK

**Pull request merged**